### ISSUE TITLE
PineconeIndexOperationClient#createIndex fails to create a new index.

### ISSUE THREAD

**Issue by tzolov** [pinecone-io/pinecone-java-client#46]
### Is this a new bug?

- [X] I believe this is a new bug
- [X] I have searched the existing issues, and I could not find an existing issue for this bug

### Current Behavior

Current attempt to create an index result into:

```
Exception in thread "main" io.pinecone.exceptions.PineconeNotFoundException: 
        at io.pinecone.exceptions.HttpErrorMapper.mapHttpStatusError(HttpErrorMapper.java:13)
        at io.pinecone.PineconeIndexOperationClient.handleResponse(PineconeIndexOperationClient.java:95)
        at io.pinecone.PineconeIndexOperationClient.createIndex(PineconeIndexOperationClient.java:62)
        at org.springframework.ai.vectorstore.Main.main(Main.java:43)
```

Note that using the same environment and credentials I can create index via the UI page as well as the https://docs.pinecone.io/reference/create_index?environment=gcp-starter.

### Expected Behavior

New `my_test_index` index should be created.

### Steps To Reproduce

Valid 

- ApiKey for FreeTier subscription 
- "gcp-starter" environment
-  project with Id: ****.
- NO EXISTING IDEXES.

```java
	public static void main(String[] args) throws IOException {

		PineconeClientConfig clientConfig = new PineconeClientConfig()
				.withApiKey("********************************")
				.withProjectName("****")
				.withEnvironment("gcp-starter");

		CreateIndexRequest createIndexRequest = new CreateIndexRequest()
				.withIndexName("my_test_index").withDimension(3);

		PineconeIndexOperationClient indexOperationClient = new PineconeIndexOperationClient(clientConfig);

		indexOperationClient.createIndex(createIndexRequest);
	}
```

### Relevant log output

```shell
Exception in thread "main" io.pinecone.exceptions.PineconeNotFoundException: 
        at io.pinecone.exceptions.HttpErrorMapper.mapHttpStatusError(HttpErrorMapper.java:13)
        at io.pinecone.PineconeIndexOperationClient.handleResponse(PineconeIndexOperationClient.java:95)
        at io.pinecone.PineconeIndexOperationClient.createIndex(PineconeIndexOperationClient.java:62)
        at org.springframework.ai.vectorstore.Main.main(Main.java:43)
```

### Environment

```markdown
- **OS**: Intel Mac OS 13.5.2 (22G91)
- **Language version**: Java 17
- **Pinecone client version**: maven dependency pinecone-client:0.6.0
```

**Comment by rohanshah18** [comment]
Thanks for reporting this issue. Due to some updates on the backend, there will be some compatibility issues in java sdk's control plane, so please expect a new release soon.

**Comment by rohanshah18** [comment]
@tzolov There should be no issues on data plane but for control plane operations, please continue to use the console.

**Comment by rohanshah18** [comment]
I have released [Java SDK v0.7.4](https://github.com/pinecone-io/pinecone-java-client/releases/tag/v0.7.4), please let me know if the new release resolves the gcp-starter env issues.

**Comment by rohanshah18** [comment]
Closing this issue since I have a confirmation that it has been resolved.
https://github.com/pinecone-io/pinecone-java-client/issues/56#issuecomment-1913551879
